cg/apps/crunchy.py

```diff
@@ -0,0 +1,320 @@
+"""
+    Module for compressing BAM to CRAM
+"""
+
+import logging
+import tempfile
+from pathlib import Path
+
+from cg.utils import Process
+from .constants import (
+    BAM_SUFFIX,
+    BAM_INDEX_SUFFIX,
+    CRAM_SUFFIX,
+    CRAM_INDEX_SUFFIX,
+    FASTQ_FIRST_READ_SUFFIX,
+    FASTQ_SECOND_READ_SUFFIX,
+    SPRING_SUFFIX,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+SBATCH_HEADER_TEMPLATE = """
+#!/bin/bash
+#SBATCH --job-name={job_name}
+#SBATCH --account={account}
+#SBATCH --ntasks={ntasks}
+#SBATCH --mem={mem}G
+#SBATCH --error={log_dir}/{job_name}.stderr
+#SBATCH --output={log_dir}/{job_name}.stdout
+#SBATCH --mail-type=FAIL
+#SBATCH --mail-user={mail_user}
+#SBATCH --time=4:00:00
+#SBATCH --qos=low
+
+set -e
+
+echo "Running on: $(hostname)"
+
+source activate {conda_env}
+
+
+"""
+
+SBATCH_BAM_TO_CRAM = """
+
+error() {{
+    if [[ -e {cram_path} ]]
+    then
+        rm {cram_path}
+    fi
+
+    if [[ -e {cram_path}.crai ]]
+    then
+        rm {cram_path}.crai
+    fi
+
+    exit 1
+}}
+
+trap error ERR
+
+crunchy -r {reference_path} -t 12 compress bam -b {bam_path} -c {cram_path}
+samtools quickcheck {cram_path}
+touch {flag_path}
+"""
+
+# SBATCH_SPRING = """
+# crunchy compress spring --first {fastq_first_path} --second {fastq_second_path} \
+# --spring-path {spring_path} --check-integrity
+# if [[ $? == 0 ]]
+# then
+#     touch {flag_path}
+# else:
+#     echo Compression failed
+# fi
+# """
+
+FLAG_PATH_SUFFIX = ".crunchy.txt"
+
+
+class CrunchyAPI:
+    """
+        API for samtools
+    """
+
+    def __init__(self, config: dict):
+
+        self.process = Process("sbatch")
+        self.slurm_account = config["crunchy"]["slurm"]["account"]
+        self.crunchy_env = config["crunchy"]["slurm"]["conda_env"]
+        self.mail_user = config["crunchy"]["slurm"]["mail_user"]
+        self.reference_path = config["crunchy"]["cram_reference"]
+
+    def bam_to_cram(self, bam_path: Path, ntasks: int, mem: int, dry_run: bool = False):
+        """
+            Compress BAM file into CRAM
+        """
+        cram_path = self.get_cram_path_from_bam(bam_path)
+        job_name = bam_path.name + "_bam_to_cram"
+        flag_path = self.get_flag_path(file_path=cram_path)
+        log_dir = bam_path.parent
+
+        sbatch_header = self._get_slurm_header(
+            job_name=job_name,
+            account=self.slurm_account,
+            log_dir=log_dir,
+            mail_user=self.mail_user,
+            conda_env=self.crunchy_env,
+            ntasks=ntasks,
+            mem=mem,
+        )
+
+        sbatch_body = self._get_slurm_bam_to_cram(
+            bam_path=bam_path,
+            cram_path=cram_path,
+            flag_path=flag_path,
+            reference_path=self.reference_path,
+        )
+
+        sbatch_content = sbatch_header + "\n" + sbatch_body
+        self._submit_sbatch(sbatch_content=sbatch_content, dry_run=dry_run)
+
+    # def spring(
```

Comment:
I suggest moving this commented code to another branch to make this merge cleaner, what do you say?

Reply:
I think not done things should not go into production, I still thinks we should remove this dangling outcommented code, what do you say?